proto/redpanda/api/console/v1alpha1/debug_bundle.proto

```diff
@@ -151,14 +246,56 @@ enum BundleErrorCode {
   // Debug bundle process not running.
   BUNDLE_ERROR_CODE_PROCESS_NOT_RUNNING = 2;
 
+  // Invalid parameters.
+  BUNDLE_ERROR_CODE_INVALID_PARAMETERS = 3;
+
+  // Debug process failed.
+  BUNDLE_ERROR_CODE_PROCESS_FAILED = 4;
+
   // Job ID not recognized.
-  BUNDLE_ERROR_CODE_INVALID_JOB_ID = 3;
+  BUNDLE_ERROR_CODE_INVALID_JOB_ID = 5;
 
   // Debug bundle process was never started.
-  BUNDLE_ERROR_CODE_PROCESS_NOT_STARTED = 4;
+  BUNDLE_ERROR_CODE_PROCESS_NOT_STARTED = 6;
 
   // Internal error.
-  BUNDLE_ERROR_CODE_INTERNAL_ERROR = 5;
+  BUNDLE_ERROR_CODE_INTERNAL_ERROR = 7;
+}
+
+enum UnhealthyReason {
+  UNHEALTHY_REASON_UNSPECIFIED = 0;
+  UNHEALTHY_REASON_NODES_DOWN = 1;
+  UNHEALTHY_REASON_LEADERLESS_PARTITIONS = 2;
+  UNHEALTHY_REASON_UNDER_REPLICATED_PARTITIONS = 3;
+  UNHEALTHY_REASON_NO_ELECTED_CONTROLLER = 4;
+  UNHEALTHY_REASON_NO_HEALTH_REPORT = 5;
+}
+
+// Request for GetClusterHealth call.
+message GetClusterHealthRequest {}
+
+// Response for GetClusterHealth call.
+message GetClusterHealthResponse {
+  // whether cluster is health or not
+  bool is_healthy = 1;
+
+  repeated UnhealthyReason unhealthy_reasons = 2;
+  int32 controller_id = 3;
+  repeated int32 all_nodes = 4;
+  repeated int32 nodes_down = 5;
+  repeated int32 nodes_in_recovery_mode = 6;
+
+  int32 leaderless_count = 7;
+  // leaderless partitions. example values:
+  map<string, PartitionIDList> leaderless_partitions = 8;
```

Comment:
@jvorcak and I had a discussion about the structure of this property and landed on this approach. I think it may be convenient for frontend to be able to just iterate on topic names and have the full list. I don't really have a strong opinion. @jvorcak if this structure is Ok with you I can change.

Reply:
Fine for me to keep then, but for a public API eventually - I think we should change it to a separate type

Reply:
The current structure is convenient for the frontend, but what @weeco suggest is also doable. If we can get consistent with the public API later, I'm ok with changing the FE for this purpose. It's up to you.